mfbt/EndianUtils.h

```diff
@@ -96,6 +101,14 @@
 #  else
 #    error "Can't handle mixed-endian architectures"
 #  endif
+#elif defined(_MSC_VER)
+// As of this writing, MSVC only targets little-endian architectures.
```

Comment:
I don't have a definitive source for this, but it should at least apply to platforms that MongoDB supports.

Additionally, the "possible implementation" of {{std::endian}} strongly implies that Windows targets expect a little-endian architecture.
https://en.cppreference.com/w/cpp/types/endian

Once we are fully transitioned to C++20, we'll have the static_assert to keep this statement honest.

Reply:
@jseyster This change looks good to me.